File: 2025/day_02/solution.py
def manufacture_gifts(gifts_to_produce: list[dict]) -> list[str]:
  result = []

  for item in gifts_to_produce:
    toy = item["toy"]
    quantity = item["quantity"]

    n = max(0, int(quantity))
    result.extend([toy] * n)

  return result

File: 2025/day_02/solution.ts
(() => {
  function manufactureGifts(
    giftsToProduce: Array<{ toy: string; quantity: number }>
  ): string[] {
    return giftsToProduce.flatMap(({ toy, quantity }) => {
      const n = Math.max(0, Math.floor(quantity));
      return n ? Array(n).fill(toy) : [];
    });
  }
})();
